### DISCUSSION TITLE
Performance good enough for lots of in & out?

### DISCUSSION THREAD

**Discussion by stefandevo** [AsamK/signal-cli#535] [Q&A]
Thanks for this tool but I was wondering what the performance is when you use it for a bot? A bot that can receive like 100 messages per minute, has to respond to it. What would be the average throughput? Any idea how many messages per second it can handle? What if you get like 40 messages per second? Will it handle all of them or would it lose messages?

sorry for all the questions.

**Comment by Atomic-Bean** [comment]
I don't think its possible to *lose* messages. As for receiving speed I am not sure, will depend on your internet connection ofc. Also incase you don't know you can run the receive command for an infinitely long time period with a negative time interval (Run `signal-cli -u +XXXXXXXXXX receive --help` for more info). So you should have one process constantly receiving and another one to send messages/do other stuff.

If you are using daemon/dbus mode then you can probably do it in the one process, but I can't confirm as I've never used it.

**Reply by Trolldemorted** [reply]
The queue on the server has an upper bound - if you don't pull the messages fast enough, they will bottle up until the queue is full, and then you will lose messages.

**Reply by Atomic-Bean** [reply]
Oh, I did not know that. Do you know what the limit is?
